testsuite/openshift/objects/gateway_api/__init__.py

```diff
@@ -145,3 +146,67 @@ def delete(self):
     @property
     def reference(self):
         return {"group": "gateway.networking.k8s.io", "kind": "Gateway", "name": self.name, "namespace": self.namespace}
+
+
+class GatewayLight(OpenShiftObject, Referencable):
+    """Lightweight Gateway object for purposes of mgc"""
+
+    @classmethod
+    def create_instance(
+        cls,
+        openshift: OpenShiftClient,
+        name: str,
+        gateway_class_name: str,
+        hostname: str,
+        placement: typing.Optional[str] = None,
+    ):
+        """Creates new instance of Gateway"""
+        labels = {}
+        if placement is not None:
+            labels = {"cluster.open-cluster-management.io/placement": placement}
+
+        model = {
+            "apiVersion": "gateway.networking.k8s.io/v1beta1",
+            "kind": "Gateway",
+            "metadata": {"name": name, "namespace": openshift.project, "labels": labels},
+            "spec": {
+                "gatewayClassName": gateway_class_name,
+                "listeners": [
+                    {
+                        "name": "api",
+                        "port": 443,
+                        "protocol": "HTTPS",
+                        "hostname": hostname,
+                        "allowedRoutes": {"namespaces": {"from": "All"}},
+                    }
+                ],
+            },
+        }
+
+        return cls(model, context=openshift.context)
+
+    def is_ready(self):
+        """Checks whether the gateway got its IP address assigned thus is ready"""
+        try:
+            addresses = self.model["status"]["addresses"]
+            multi_cluster_addresses = [
+                address for address in addresses if address["type"] == "kuadrant.io/MultiClusterIPAddress"
+            ]
+            return len(multi_cluster_addresses) > 0
+        except (KeyError, ModelError):
+            return False
+
+    @backoff.on_predicate(backoff.fibo, predicate=lambda x: not x.is_ready(), max_tries=20, jitter=None)
+    def wait_for_ready(self):
+        """Waits for the gateway to be ready in the sense of is_ready(self)"""
+        self.refresh()
+        return self
```

Comment:
You can use until_all method from openshift library instead https://github.com/Kuadrant/testsuite/blob/5a67e31174be1aa6ace7fa3ba7c5922e067637a1/testsuite/openshift/objects/authorino.py#L50